src/app/config/config.c

```diff
@@ -2994,14 +2997,32 @@ resolve_my_address(int warn_severity, const or_options_t *options,
 MOCK_IMPL(int,
 is_local_addr, (const tor_addr_t *addr))
 {
-  if (tor_addr_is_internal(addr, 0))
+  int EnforceDistinctSubnets = get_options()->EnforceDistinctSubnets;
+
+  if (tor_addr_family(addr) == AF_INET6) {
```

Comment:
is_local_addr() should be a simple wrapper for is_local_addr_impl().

It should contain a single call to is_local_addr_impl(), like this:
`is_local_addr_impl(EnforceDistinctSubnets, addr, last_resolved_addr, valid_ipv6_addr);`

valid_ipv6_addr should be:
* `&(router_get_my_routerinfo()->ipv6_addr)` if it is valid
* NULL if it is not valid

You should not need to do any checks on other_addr.